MdePkg/Include/Protocol/NvmeOFPassthru.h

```diff
@@ -34,8 +39,8 @@ typedef struct _NVMEOF_CONNECT_COMMAND {
 } NVMEOF_CONNECT_COMMAND;
 
 typedef struct _NVMEOF_CLI_CTRL_MAPPING {
-  struct spdk_nvme_ctrlr   *Ctrlr;
-  struct spdk_nvme_qpair   *Ioqpair;
+  VOID                     *Ctrlr;
```

Comment:
Why don't you simply declare `struct spdk_nvme_ctrlr` above? Using VOID for something that's going to hold a specific struct pointer is not beautiful.

Reply:
More such cases below.

Reply:
Unless I am mistaken, the point of these changes are to remove all SPDK specific data structures and definitions from this part of the code.  These `(void *)` definitions are needed to abstract SPDK from the EDK2 protocol layer.

Reply:
I understand. Yet I wonder if we can't find some way to do this without using VOID, enabling at least some very basic type checking by the compiler. Would upstream EDK2 object an empty struct declaration like this?

```
struct spdk_nvme_ctrlr;
```

If so, maybe we could do something like this in the header file:

```
struct edk2_nvme_ctrl_impl;
```

and then in the actual NVMe code we could simply do this:

```
#define edk2_nvme_ctrl_impl spdk_nvme_ctrlr
```

And then see the strange cast below, am I misreading that code?

Reply:
I suppose using an empty struct definition with a `#define` could work, but I don't see the point.  The reason for this change is because they want the network driver abstracted and undefined.  This code should support any network driver or controller... not just SPDK.  And adding an empty struct definition doesn't do that.  If EDK is anything like SPDK all exported header files must be independently compiled.   

Personally I don't see any problems with a `VOID`. 

@amit-jain9 should say more.